### PR TITLE
enumerable methods done - review

### PR DESCRIPTION
**Pull request by rOluochKe**

### CODE REVIEW

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    #none
+    def my_none?
+        result = true
+        self.my_each {|i| result = false if yield(i)}
+        result
+    end
+
+    #count
+    def my_count
```
Comment (bolah2009) [inline enumerable.rb]: #### `REQUIRED`

### :bug: Let's add a case for when a block is not given. Compare with the built-in method. Learn more [here](https://ruby-doc.org/core-2.6.3/Enumerable.html#method-i-count)

#### 1. When `arg` is not a `block`
```ruby
$ irb
irb(main):001:0> require_relative 'enumerable'
=> true
irb(main):002:0> [1,2,3,4,5].count(3)
=> 1
irb(main):003:0> [1,2,3,4,5].my_count(3)
Traceback (most recent call last):
        5: from /../.rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        4: from /../.rbenv/versions/2.6.3/bin/irb:23:in `load'
        3: from /../.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):3
        1: from /../enumerable.rb:51:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
```

#### 2. When no `arg` is given

```ruby
irb(main):004:0> [1,2,3,4,5].count
=> 5
irb(main):005:0> [1,2,3,4,5].my_count
Traceback (most recent call last):
        7: from /../.rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        6: from /../.rbenv/versions/2.6.3/bin/irb:23:in `load'
        5: from /../.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        4: from (irb):5
        3: from /../enumerable.rb:53:in `my_count'
        2: from /../enumerable.rb:8:in `my_each'
        1: from /../enumerable.rb:53:in `block in my_count'
LocalJumpError (no block given (yield))
irb(main):006:0> 
```

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    def my_each
+        i = 0
+        while i < self.length
+            yield(self[i])
+            i += 1
+        end
+    end
+
+    #each_with_index
+    def my_each_with_index
```
Comment (bolah2009) [inline enumerable.rb]: #### `REQUIRED`

### :bug: Method not working as expected, the `index` value is not captured as illustrated below. Compare with the built-in method. Learn more [here](https://ruby-doc.org/core-2.6.3/Enumerable.html#method-i-each_with_index)

```ruby
$ irb
irb(main):001:0> require_relative 'enumerable'
=> true
irb(main):002:0> hash = {}
=> {}
irb(main):003:0> %w(cat dog rat).each_with_index { |item, index| hash[item] = index }
=> ["cat", "dog", "rat"]
irb(main):004:0> hash
=> {"cat"=>0, "dog"=>1, "rat"=>2}
irb(main):005:0> %w(cat dog rat).my_each_with_index { |item, index| hash[item] = index }
=> nil
irb(main):006:0> hash
=> {"cat"=>0, "dog"=>0, "rat"=>0}
```

File: enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        self.my_each {|i| result = true if yield(i)}
+        result
+    end
+
+    #none
+    def my_none?
+        result = true
+        self.my_each {|i| result = false if yield(i)}
+        result
+    end
```
Comment (maelfosso) [inline enumerable.rb]: Sorry it's no working when there is no **`block`**
```ruby
irb(main):024:0> [nil, false].my_none? 
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):24
        3: from enumerable.rb:46:in `my_none?'
        2: from enumerable.rb:8:in `my_each'
        1: from enumerable.rb:46:in `block in my_none?'
LocalJumpError (no block given (yield))
irb(main):025:0> [nil, false, true].none?
=> false
irb(main):026:0> [nil, false, true].my_none?
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):26
        3: from enumerable.rb:46:in `my_none?'
        2: from enumerable.rb:8:in `my_each'
        1: from enumerable.rb:46:in `block in my_none?'
LocalJumpError (no block given (yield))
```

File: enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+    #count
+    def my_count
+        total = 0
+        self.my_each{total += 1}
+        total
+    end
```
Comment (maelfosso) [inline enumerable.rb]: Sorry but it's still not working.
```ruby
irb(main):030:0> ary.count(2)
=> 2
irb(main):031:0> ary.my_count(2)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):31
        1: from enumerable.rb:51:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):032:0> ary.count{ |x| x%2==0 }
=> 3
irb(main):033:0> ary.my_count{ |x| x%2==0 }
=> 4
irb(main):034:0> 
```

File: enumerable.rb [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        self.my_each {|i| result.push(block.call(i))}
+        return result
+    end
+
+    #inject
+    def my_inject
+        m = self[0]
+        self.my_each {|i| m = yield(m, i)}
+        m
+    end
```
Comment (maelfosso) [inline enumerable.rb]: Sorry it's not working as expected
```ruby
irb(main):034:0> [5,6,7,8,9,10].inject(1) { |product, n| product * n }
=> 151200
irb(main):035:0> [5,6,7,8,9,10].my_inject(1) { |product, n| product * n }
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):35
        1: from enumerable.rb:65:in `my_inject'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):036:0> [5,6,7,8,9,10].inject { |product, n| product * n }
=> 151200
irb(main):037:0> [5,6,7,8,9,10].my_inject { |product, n| product * n }
=> 756000
irb(main):038:0> [5,6,7,8,9,10].inject { |sum, n| sum + n }
=> 45
irb(main):039:0> [5,6,7,8,9,10].my_inject { |sum, n| sum + n }
=> 50
irb(main):040:0> [5,6,7,8,9,10].inject(5) { |sum, n| sum + n }
=> 50
irb(main):041:0> [5,6,7,8,9,10].my_inject(5) { |sum, n| sum + n }
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):41
        1: from enumerable.rb:65:in `my_inject'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):042:0> 
```

File: enumerable.rb [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+end
+
+
+#==================================TESTING INSTRUCTIONS============================
+#Each test is organized in order, uncomment the test block to run the test you need
+#==================================================================================
+
+
+#=========== MY EACH METHOD TEST ======================
+# array=[6,4,8,14,56,12,4,5,8]
```
Comment (bolah2009) [inline enumerable.rb]: #### `REQUIRED`

Consider deleting commented out codes. If you need to review a previous version of your code, use git history.


You can move this to the `README` to further explain what the project does. :blush: 
#### Nice documentation. :ok:

File: bin/main.rb [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+# frozen_string_literal: true
+
+module Enumerable
+  # my_each method
+  def my_each
+    counter = 0
+    while counter < size
+      yield(self[counter])
+      counter += 1
+    end
```
Comment (Oghenebrume50) [inline bin/main.rb]: What happens when no block is given?

File: bin/main.rb [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+      end
+    else
+      for i in self
+        if yield(i)
+          count += 1
+        end
+      end
+    end
+    count
+  end
```
Comment (Oghenebrume50) [inline bin/main.rb]: [OPTIONAL] Favour an each loop over a for loop see [this](https://stackoverflow.com/questions/20198197/is-there-any-case-to-prefer-for-instead-of-each-in-ruby)

**Pull request merged**